Universal FE Randomizer/src/random/gba/randomizer/shuffling/CharacterShuffler.java

```diff
@@ -0,0 +1,325 @@
+				int targetClassId =  GBACrossGameData.getEquivalentClass(type, crossGameData).getID();
+
+				if (targetClassId == 0) {
+					DebugPrinter.error(DebugPrinter.Key.GBA_CHARACTER_SHUFFLING, String.format("Couldn't find an applicable class for character %s class %s%n", crossGameData.name,
+							crossGameData.characterClass));
+					continue;
+				}
+				
+				// (b) initialize some variables needed later
+				GBAFEClassData targetClass = classData.classForID(targetClassId);
+				GBAFEClassData sourceClass = classData.classForID(slot.getClassID());
+				int slotLevel = slot.getLevel();
+
+				// (c) Insert the portrait for the current character into the rom and repoint the Portrait Data
+				try {
+					changePortrait(slot, portraitData, type, crossGameData, freeSpace, fileHandler);
+				} catch (Exception e) {
+					e.printStackTrace();
+					continue;
+				}
+				
+				// (d) Update name and Description
+				updateName(textData, portraitData, slot, crossGameData);
+				
+				// Give an option to not change the description to help with keeping track of which character is which
+				if (options.shouldChangeDescription()) {
+					// [0x1] = new line 
+					// [X] = end of text segment?
+					textData.setStringAtIndex(slot.getDescriptionIndex(),String.format("%s[0x1]%s[X]", crossGameData.description1, crossGameData.description2));
+				}
+				
+				for (GBAFECharacterData linkedSlot : characterData.linkedCharactersForCharacter(slot)) {
+					linkedSlot.setGrowths(crossGameData.growths);
+					linkedSlot.setConstitution(crossGameData.constitution);
+					updateWeaponRanks(linkedSlot, crossGameData, options);
+					
+					// (e) Update the bases, and potentially auto level the Character to the level of the slot.
+					// Due to Promotion / Demotion, the output of the targetClass might be different from what was passed into this method
+					targetClass = updateBases(textData, rng, classData, options, linkedSlot, crossGameData, targetClassId,
+							targetClass, sourceClass, slotLevel);
+					targetClassId = targetClass.getID();
+	
+					
+					// (f) Update the class for all the slots of the character
+					updateUnitInChapter(chapterData, linkedSlot, crossGameData, targetClassId, options, itemData, inventoryOptions);
+						
+					// (g) give the Unit new items to use
+					ItemAssignmentService.assignNewItems(characterData, linkedSlot, targetClass, chapterData, inventoryOptions, rng, textData, classData, itemData);
+				}
+			}
+
+		}
+	}
+
+	
+	private static void updateName(TextLoader textData, PortraitDataLoader portraitData,
+			GBAFECharacterData slot, GBACrossGameData crossGameData) {
+		List<Integer> nameIndicies = new ArrayList<>();
+		nameIndicies.add(slot.getNameIndex());
+		nameIndicies.addAll(portraitData.getRelatedNameIndicies(slot.getNameIndex()));
+		for(Integer index : nameIndicies) {
+			textData.setStringAtIndex(index, crossGameData.name+"[X]");
+		}
```

Comment:
Might be worth lifting the logic from the old RecruitmentRandomizer to do a name change across the entire script (unless you do it later, I haven't gotten that far yet).

Reply:
I missed that so far, I'll take a look into making this reusable